core/workspace_comment.js

```diff
@@ -0,0 +1,370 @@
+   * @type {!goog.math.Coordinate}
+   * @protected
+   */
+  this.xy_ = new goog.math.Coordinate(0, 0);
+
+  /**
+   * The comment's height in workspace units.  Scale does not change this value.
+   * @type {number}
+   * @private
+   */
+  this.height_ = height;
+
+  /**
+   * The comment's width in workspace units.  Scale does not change this value.
+   * @type {number}
+   * @private
+   */
+  this.width_ = width;
+
+  /**
+   * @type {!Blockly.Workspace}
+   */
+  this.workspace = workspace;
+
+  /**
+   * @protected
+   * @type {boolean}
+   */
+  this.RTL = workspace.RTL;
+
+  /**
+   * @type {boolean}
+   * @private
+   */
+  this.deletable_ = true;
+
+  /**
+   * @type {boolean}
+   * @private
+   */
+  this.movable_ = true;
+
+  /**
+   * @protected
+   * @type {!string}
+   */
+  this.content_ = content;
+
+  /**
+   * @package
+   * @type {boolean}
+   */
+  this.isComment = true;
+
+  Blockly.WorkspaceComment.fireCreateEvent(this);
+};
+
+/**
+ * Dispose of this comment.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.dispose = function() {
+  if (!this.workspace) {
+    // The comment has already been deleted.
+    return;
+  }
+
+  if (Blockly.Events.isEnabled()) {
+    Blockly.Events.fire(new Blockly.Events.CommentDelete(this));
+  }
+
+  // Remove from the list of top comments and the comment database.
+  this.workspace.removeTopComment(this);
+  this.workspace = null;
+};
+
+// Height, width, x, and y are all stored on even non-rendered comments, to
+// preserve state if you pass the contents through a headless workspace.
+
+/**
+ * Get comment height.
+ * @return {number} comment height.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.getHeight = function() {
+  return this.height_;
+};
+
+/**
+ * Set comment height.
+ * @param {number} height comment height.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.setHeight = function(height) {
+  this.height_ = height;
+};
+
+/**
+ * Get comment width.
+ * @return {number} comment width.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.getWidth = function() {
+  return this.width_;
+};
+
+/**
+ * Set comment width.
+ * @param {number} width comment width.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.setWidth = function(width) {
+  this.width_ = width;
+};
+
+/**
+ * Get stored location.
+ * @return {!goog.math.Coordinate} The comment's stored location.  This is not
+ *     valid if the comment is currently being dragged.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.getXY = function() {
+  return this.xy_.clone();
+};
+
+/**
+ * Move a comment by a relative offset.
+ * @param {number} dx Horizontal offset, in workspace units.
+ * @param {number} dy Vertical offset, in workspace units.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.moveBy = function(dx, dy) {
+  var event = new Blockly.Events.CommentMove(this);
+  this.xy_.translate(dx, dy);
+  event.recordNew();
+  Blockly.Events.fire(event);
```

Comment:
Similar to above, should there be an events isEnabled check here?

Reply:
Doesn't need one--it checks inside `fire`.  Normally we use that check to avoid constructing the event itself if it's unnecessary.  No harm is done by avoiding it.